DirectEthernet/src/main/scala/org/labrad/ethernet/server/EthernetBus.scala

```diff
@@ -58,26 +59,84 @@ class EthernetBus(
   private lazy val listenThread = {
     val thread = new Thread {
       override def run(): Unit = {
-        try {
-          pcap.loop(-1, packetListener)
-          pcap.close()
-        } catch {
-          case e: Exception =>
-            // should never get here
-            e.printStackTrace()
-            throw e
+        var alive = true
+        while (alive) {
+          val pcap = reopen()
+          try {
+            pcap.loop(-1, packetListener)
```

Comment:
yes, nominally forever. That's why we spin up a separate listener thread for each ethernet bus.